File: backend/src/main/java/group8/bloodbank/model/MedicalWorker.java
package group8.bloodbank.model;

import com.fasterxml.jackson.annotation.JsonIgnore;
import lombok.AllArgsConstructor;
import org.hibernate.annotations.Fetch;
import org.hibernate.annotations.FetchMode;

import javax.persistence.*;
import java.util.List;
import java.util.Set;

@Entity
@AllArgsConstructor
public class MedicalWorker extends User {


    public MedicalWorker(Long id, String name, String surname, String password, Address adress, String jmbg, String email, String occupation, Gender gender) {
        super(id, name, surname, password, adress, jmbg, email, occupation, gender, UserType.MEDICAL_WORKER);
    }

    public MedicalWorker(String name, String surname, String email, String password, String jmbg, Address address, String occupation,  Gender gender, BloodBank bb) {
        super(name, surname, email, password, jmbg, address, occupation, gender, UserType.MEDICAL_WORKER);
        this.bloodBank = bb;
    }

    public MedicalWorker(String name, String surname, String email, String password, String jmbg, Address address, String occupation,  Gender gender) {
        super(name, surname, email, password, jmbg, address, occupation, gender, UserType.MEDICAL_WORKER);

    }

    public MedicalWorker() {

    }

    @ManyToOne
    @JoinColumn(name = "blood_bank_id")
    private BloodBank bloodBank;

    @ManyToMany(fetch = FetchType.EAGER, cascade = CascadeType.ALL)
    @JoinTable(
            name = "medical_worker_appointments",
            joinColumns = @JoinColumn(name = "medical_worker_id"),
            inverseJoinColumns = @JoinColumn(name = "appointment_id"))
    private Set<Appointment> appointments;

    public MedicalWorker(Long id, String name, String surname, String email, String password, String jmbg, Address address, String occupation, Gender gender) {
        super(name, surname, email, password, jmbg, address, occupation, gender, UserType.MEDICAL_WORKER);
        setId(id);
    }

    @JsonIgnore
    @OneToMany(mappedBy = "medicalWorker", fetch = FetchType.EAGER)
    @Fetch(value = FetchMode.SUBSELECT)
    private List<AppointmentHistory> appointmentHistories;

    public BloodBank getBloodBank() {
        return bloodBank;
    }

    public void setBloodBank(BloodBank newBloodBank) {
        this.bloodBank = newBloodBank;
    }
}

File: frontend/src/app/model/medicalWorker.ts
import { BloodBank } from './bloodBank';
import { Address } from './address';
export class MedicalWorker {
    name: string = '';
    surname: string = '';
    password: string = '';
    jmbg: string = '';
    email: string = '';
    occupation: string ='';
    gender: string = '';
    address!: Address;
    bloodBank!: BloodBank;
    id: number = 0;

  public constructor(obj?: any) {
      if (obj) {
          this.id = obj.id;
          this.name = obj.name;
          this.surname = obj.surname;
          this.password = obj.password;
          this.jmbg = obj.jmbg;
          this.email = obj.email;
          this.occupation = obj.occupation;
          this.gender = obj.gender;
          this.address = obj.address;
      }
  }

  public fullName(): String {
    return this.name + ' ' + this.surname;
  }
}
